spec/classes/otelcol_spec.rb

```diff
@@ -274,61 +263,51 @@
             },
           }
         end
-        let(:configcontent_ext) do
-          configcontent.merge(
-            {
-              'extensions' => { 'test' => {} },
-              'service' => {
-                'extensions' => ['test'],
-                'pipelines' => {},
-                'telemetry' => {
-                  'logs' => {},
-                  'metrics' => {
-                    'level' => 'basic',
-                    'address' => ':8888',
-                  },
-                },
-              }
-            }
-          )
-        end
 
         it { is_expected.to compile.with_all_deps }
-        it { is_expected.to contain_file('otelcol-config').with_content(configcontent_ext.to_yaml) }
-      end
-
-      context 'with include files' do
-        let :params do
-          {
-            processors: '${file:processors.yaml}',
-            exporters: '${file:exporters.yaml}',
-            pipelines: '${file:pipelines.yaml}',
-          }
-        end
-        let(:configcontent_ext) do
-          configcontent.merge(
-            {
-              'processors' => '${file:processors.yaml}',
-              'exporters' => '${file:exporters.yaml}',
-              'service' => {
-                'extensions' => [],
-                'pipelines' => '${file:pipelines.yaml}',
-                'telemetry' => {
-                  'logs' => {},
-                  'metrics' => {
-                    'level' => 'basic',
-                    'address' => ':8888',
-                  },
-                },
-              }
-            }
-          )
-        end
 
-        it { is_expected.to compile.with_all_deps }
-        it { is_expected.to contain_file('otelcol-config').with_content(configcontent_ext.to_yaml) }
+        it {
+          is_expected.to contain_otelcol__extension('test').with({
+                                                                   'config' => {},
+                                                                   'order' => 0,
+                                                                   'name' => 'test',
+                                                                 })
+          is_expected.to contain_concat__fragment('otelcol-config-extension-test')
+        }
       end
 
+      # context 'with include files' do
```

Comment:
Generally not a fan of commented code. Either make it work, or drop it. I assume this one fails because the YAML isn't exactly the same.

You can get the content with `catalog.resource('file', 'otelcol-config').send(:parameters)[:content]`, so this probably works:
```ruby
it do
  is_expected.to contain_file('otelcol-config')
  expect(YAML.safe_load(catalog.resource('file', 'otelcol-config').send(:parameters)[:content])).to eq(configcontent_ext)
end
```

Reply:
Removed that code